mindmeld/components/nlp.py

```diff
@@ -1674,18 +1675,18 @@ def unload(self):
         self.role_classifier.unload()
 
     def _load(self, incremental_timestamp=None):
+        model_path, incremental_model_path = path.get_role_model_paths(
+            self._app_path,
+            self.domain,
+            self.intent,
+            self.type,
+            timestamp=incremental_timestamp,
+        )
+        self.role_classifier.load(
+            incremental_model_path if incremental_timestamp else model_path
+        )
         try:
-            model_path, incremental_model_path = path.get_role_model_paths(
-                self._app_path,
-                self.domain,
-                self.intent,
-                self.type,
-                timestamp=incremental_timestamp,
-            )
-            self.role_classifier.load(
-                incremental_model_path if incremental_timestamp else model_path
-            )
-            self.entity_resolver.load()
+            self.entity_resolver.fit(clean=False)
```

Comment:
Just wonder do we need this for backward compatibility reasons? I think in general we would prefer not to do `fit()` when loading the models so that the behavior across NLP pipeline is more consistent and predictable. (so that users won't be surprised by any kind of model building happens during `load()`) I think the changes in this PR help us move forward on to that direction since we don't need to do fit() in the resolver implementation of load() anymore and we should keep it that way if we can?

Reply:
Yeah, I absolutely agree to you point Marvin. I did so keeping in mind about the WxA; I realized that if we dump and load resolver models (exact match in WxA case), we will be using more disk space as we need to dump the entity map (i.e. a processed KB data) in the dump() call and load it back in the load() call of NLP pipeline. So I would like to confirm if extra disk space is a problem in case of WxA before making the change you suggested. @vijay120 @vembar @mhuang2173

Reply:
Update: It is more logical to not to dump a copy of the KB data when we call dump() for a resolver. In the latest push, I added 'entity_map' argument in the load() method to address this. So what happens now is that, the KB data is loaded in the load() method as well similar to fit() method and hence I can replace `'self.entity_resolver.fit(clean=False)'`with `'self.entity_resolver.load()'`. Also in case of question answerers, we can pass the KB data looked up the question answerers directly to the load() method.